### ISSUE TITLE
Corrupted file

### ISSUE THREAD

**Issue by Woffelson** [IsmAvatar/LateralGM#246]
LGM gave this error report when I opened a file that was saved in LGM. Now the project file doesn't work. (Luckily I have the original project as a backup.)

Operating System: Linux
Version: 3.13.0-74-generic
Architecture: amd64

Java Vendor: Oracle Corporation
Version: 1.7.0_91

Available processors (cores): 2
Free memory (bytes): 79036352
Maximum memory (bytes): 1392508928
Total memory available to JVM (bytes): 120061952

File system root: /
Total space (bytes): 236154146816
Free space (bytes): 40357412864
Usable space (bytes): 28337848320

Stack trace:
org.lateralgm.file.GmFormatException: java.net.MalformedURLException: no protocol: /media/arello/B4DB-04ED/Pelejä/Enigma/Toimiva plattis.gmx/sprites/spr_p3.sprite.gmx
	at java.net.URL.<init>(URL.java:589)
	at java.net.URL.<init>(URL.java:486)
	at java.net.URL.<init>(URL.java:435)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:619)
	at com.sun.org.apache.xerces.internal.impl.XMLVersionDetector.determineDocVersion(XMLVersionDetector.java:189)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:812)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777)
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:243)
	at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:338)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:177)
	at org.lateralgm.file.GMXFileReader.iterateSprites(GMXFileReader.java:542)
	at org.lateralgm.file.GMXFileReader.readSprites(GMXFileReader.java:615)
	at org.lateralgm.file.GMXFileReader.readProjectFile(GMXFileReader.java:244)
	at org.lateralgm.file.GMXFileReader.readProjectFile(GMXFileReader.java:201)
	at org.lateralgm.main.FileChooser$ProjectReader.read(FileChooser.java:383)
	at org.lateralgm.main.FileChooser$1.run(FileChooser.java:564)
	at java.lang.Thread.run(Thread.java:745)

**Comment by RobertBColton** [comment]
Possible duplicate of #229 can you upload/send the corrupted file?

It looks like it's saying there is a sprite that isn't there, so the sprite is in the main tree but the actual sprite data file is missing. That's what it looks like at first sight anyway.

**Comment by Woffelson** [comment]
D'oh, I already deleted the file since I made numerous try-outs. I also can't open the original file anymore even though I didn't overwrite or save it at any point, so I'm afraid I can't replicate this unless I try this with another project...

**Comment by RobertBColton** [comment]
@Woffelson I won't be able to fix it if we can't figure out the specific issue and make it reproducible, where I can consistently make it occur. You do still have that specific game that caused it though, the original, perhaps try loading it in LGM and saving an extra copy to see if it happens again. If it can't be reproduced consistently, I will have to close this until it can be or more information is supplied.

**Comment by Woffelson** [comment]
Well, I couldn't reproduce the bug identically since the saved project didn't corrupt this time. However, when I saved the project earlier in EGM-format, it gave this following error. The file is [here](http://files.1337upload.net/plattis-77140f.zip).

Operating System: Linux
Version: 3.13.0-74-generic
Architecture: amd64

Java Vendor: Oracle Corporation
Version: 1.7.0_91

Available processors (cores): 2
Free memory (bytes): 121153056
Maximum memory (bytes): 1392508928
Total memory available to JVM (bytes): 171442176

File system root: /
Total space (bytes): 236154146816
Free space (bytes): 39917527040
Usable space (bytes): 27897962496

Stack trace:
java.lang.NullPointerException
	at org.enigma.file.EFileReader.readResource(EFileReader.java:666)
	at org.enigma.file.EFileReader.processEntries(EFileReader.java:644)
	at org.enigma.file.EFileReader.readNodeChildren(EFileReader.java:622)
	at org.enigma.file.EFileReader.readEgmFile(EFileReader.java:560)
	at org.enigma.file.EFileReader.readEgmFile(EFileReader.java:538)
	at org.enigma.file.EgmIO.read(EgmIO.java:56)
	at org.lateralgm.main.FileChooser$1.run(FileChooser.java:564)
	at java.lang.Thread.run(Thread.java:745)

**Comment by RobertBColton** [comment]
I just fixed this in #316, apparently parse actually takes a URI and I was handing it a raw file path with no protocol. Basically a file path is "C:/Users/Owner/..." and a URI is "file:/C:/Users/Owner" but this for some reason was never problem on Windows. Regardless, it is now fixed properly and I pass the parse method a File object constructed from the file path. The fix will be included in the new LGM release tonight.

I have moved the EGM issue to the plugin tracker since that is not an issue with LGM directly, but rather an issue with the EGM reader in its ENIGMA plugin.
https://github.com/enigma-dev/lgmplugin/issues/50
